File: src/infrastructure/ProtobuildException.php
<?php

final class ProtobuildException extends Exception {
  
  private $protobuildMessage;
  
  public function __construct($message) {
    $this->protobuildMessage = $message;
  }
  
  public function getProtobuildMessage() {
    return $this->protobuildMessage;
  }
  
}

File: src/Protobuild.Website/Exceptions/ProtobuildException.cs
using System.Net;

namespace Protobuild.Website.Exceptions
{
    public class ProtobuildException : HttpException
    {
        public ProtobuildException(HttpStatusCode code, string message) : base(code, message)
        {
            ProtobuildMessage = message;
        }

        public string ProtobuildMessage { get; }
    }
}
